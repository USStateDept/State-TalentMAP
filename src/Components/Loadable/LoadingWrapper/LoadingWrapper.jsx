import PropTypes from 'prop-types';
import LoadingError from '../LoadingError';
import Spinner from '../../Spinner';

const LoadingWrapper = (props) => {
  if (props.error) {
    return (
      <LoadingError />
    );
  }
  return props.placeholder || <Spinner size={props.size} />;
};

LoadingWrapper.propTypes = {
  error: PropTypes.shape({}),
  size: PropTypes.oneOf(['big', 'small']),
  placeholder: PropTypes.node,
};

LoadingWrapper.defaultProps = {
  error: undefined,
  size: 'big',
  placeholder: null,
};

export default LoadingWrapper;
