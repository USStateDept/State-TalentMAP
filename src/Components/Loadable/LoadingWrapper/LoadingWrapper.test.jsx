import { shallow } from 'enzyme';
import React from 'react';
import toJSON from 'enzyme-to-json';
import LoadingWrapper from './LoadingWrapper';

describe('LoadingWrapper', () => {
  it('is defined', () => {
    const wrapper = shallow(<LoadingWrapper />);
    expect(wrapper).toBeDefined();
  });

  it('is renders LoadingError when error prop exists', () => {
    const wrapper = shallow(<LoadingWrapper error={{}} />);
    expect(wrapper.find('LoadingError').exists()).toBe(true);
  });

  it('matches snapshot', () => {
    const wrapper = <LoadingWrapper />;
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
