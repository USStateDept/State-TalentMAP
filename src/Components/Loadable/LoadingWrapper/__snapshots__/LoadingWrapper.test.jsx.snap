// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingWrapper matches snapshot 1`] = `undefined`;
